src/wallet/message_store.cpp

```diff
@@ -661,29 +709,30 @@ void message_store::delete_all_messages()
   m_messages.clear();
 }
 
-// Make a message text, which is "attacker controlled data", reasonably safe to display
+// Make a text, which is "attacker controlled data", reasonably safe to display
 // This is mostly geared towards the safe display of notes sent by "mms note" with a "mms show" command
-void message_store::get_sanitized_message_text(const message &m, std::string &sanitized_text) const
+void message_store::get_sanitized_text(const std::string &text, size_t max_length, std::string &sanitized_text) const
 {
   sanitized_text.clear();
 
   // Restrict the size to fend of DOS-style attacks with heaps of data
-  size_t length = std::min(m.content.length(), (size_t)1000);
+  size_t length = std::min(text.length(), max_length);
 
   for (size_t i = 0; i < length; ++i)
   {
-    char c = m.content[i];
-    if ((int)c < 32)
+    char c = text[i];
+    unsigned int code = (unsigned int)c;
+    if ((code < 0x20) || (code == 0x7f) || (code >= 0x80 && code <= 0x9f))
```

Comment:
That will probably mangle UTF-8 with more than one byte. You want to perform the range checks on the code points, not the individual bytes. See the log sanitization code in my last patch (and you can use tools::utf8canonical to parse it for you).

Reply:
Done. I have a very small residual problem left that cutting the string at max length can cut through an UTF-8 sequence and make the whole string invalid that way, but that problem can be ignored for now.